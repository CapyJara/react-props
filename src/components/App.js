import React from 'react';
import Color from './Color';


export default function App() {
  const name = 'Pearl Violet';
  const rgb = {
    red: 134,
    green: 115,
    blue: 161
  };
  const hex = '8673a1';
  
  return <Color name={name} rgb={rgb} hex={hex} />;
}
