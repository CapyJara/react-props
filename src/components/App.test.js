import React from 'react';
import { shallow } from 'enzyme';
import App from './App';

describe('Dog component', () => {
  it('renders a pink with all fields', () => {
    const wrapper = shallow(<App name="Pink" rgb={ { red: 255, green: 105, blue: 180 } } hex="ff69b4" />);
    expect(wrapper).toMatchSnapshot();
  });
  it('renders a pink without hex', () => {
    const wrapper = shallow(<App name="Pink" rgb={ { red: 255, green: 105, blue: 180 } } />);
    expect(wrapper).toMatchSnapshot();
  });
  it('renders a pink without name', () => {
    const wrapper = shallow(<App rgb={ { red: 255, green: 105, blue: 180 } } />);
    expect(wrapper).toMatchSnapshot();
  });
});
