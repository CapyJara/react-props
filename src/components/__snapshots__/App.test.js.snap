// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dog component renders a pink with all fields 1`] = `
<Color
  hex="8673a1"
  name="Pearl Violet"
  rgb={
    Object {
      "blue": 161,
      "green": 115,
      "red": 134,
    }
  }
/>
`;

exports[`Dog component renders a pink without hex 1`] = `
<Color
  hex="8673a1"
  name="Pearl Violet"
  rgb={
    Object {
      "blue": 161,
      "green": 115,
      "red": 134,
    }
  }
/>
`;

exports[`Dog component renders a pink without name 1`] = `
<Color
  hex="8673a1"
  name="Pearl Violet"
  rgb={
    Object {
      "blue": 161,
      "green": 115,
      "red": 134,
    }
  }
/>
`;
